src/backend/services/request_validators.py

```diff
@@ -240,16 +308,56 @@ async def validate_update_agent_request(session: DBSessionDep, request: Request)
             detail=f"If updating an agent's model, the deployment must also be provided.",
         )
     elif model and deployment:
-        # Validate deployment
-        if deployment not in AVAILABLE_MODEL_DEPLOYMENTS.keys():
-            raise HTTPException(
-                status_code=400,
-                detail=f"Deployment {deployment} not found or is not available.",
-            )
+        validate_deployment_model(deployment, model, session)
+
+
+async def validate_create_update_model_request(session: DBSessionDep, request: Request):
+    """
+    Validate that the create model request has valid deployment.
+
+    Args:
+        request (Request): The request to validate
+
+    Raises:
+        HTTPException: If the request does not have the appropriate values in the body
+    """
+    body = await request.json()
+    deployment_id = body.get("deployment_id")
+    if request.method == "POST" and not deployment_id:
+        raise HTTPException(status_code=400, detail="deployment_id is required.")
 
-        # Validate model
-        if model not in AVAILABLE_MODEL_DEPLOYMENTS[deployment].models:
+    if deployment_id:
+        deployment_db = deployment_crud.get_deployment(session, deployment_id)
+        if not deployment_db:
             raise HTTPException(
                 status_code=400,
-                detail=f"Model {model} not found for deployment {deployment}.",
+                detail=f"Deployment {deployment_id} not found or is not available in the Database.",
             )
+
+
+async def validate_create_deployment_request(session: DBSessionDep, request: Request):
+    """
+    Validate that the create deployment request has valid env vars.
```

Comment:
Possibly the docstrings here are wrong? It doesn't seem like env vars are being validated here

Reply:
Ty